tests/output.rs

```diff
@@ -0,0 +1,174 @@
+use assert_cmd::prelude::*;
+use once_cell::sync::Lazy;
+use predicates::{prelude::*, BoxPredicate};
+use std::{
+    env,
+    ffi::OsStr,
+    path::{Path, PathBuf},
+    process::Command,
+};
+
+struct Location {
+    line: usize,
+    #[allow(dead_code)]
+    column: usize,
+}
+
+impl From<(usize, usize)> for Location {
+    fn from(value: (usize, usize)) -> Self {
+        Self {
+            line: value.0,
+            column: value.1,
+        }
+    }
+}
+
+struct Range {
+    pub start: Location,
+    #[allow(dead_code)]
+    pub end: Location,
+}
+
+impl From<((usize, usize), (usize, usize))> for Range {
+    fn from(value: ((usize, usize), (usize, usize))) -> Self {
+        Self {
+            start: value.0.into(),
+            end: value.1.into(),
+        }
+    }
+}
+
+struct ExpectedMatch {
+    pub relative_file_path: String,
+    pub lines: Vec<String>,
+    pub range: Range,
+}
+
+impl ExpectedMatch {
+    pub fn expected_output_text(&self) -> String {
+        self.lines
+            .iter()
+            .enumerate()
+            .map(|(line_index, line)| {
+                format!(
+                    "{}:{}:{}",
+                    self.relative_file_path,
+                    self.range.start.line + line_index + 1,
+                    line
+                )
+            })
+            .collect::<Vec<_>>()
+            .join("\n")
```

Comment:
Since the current default(/only) output format shows all of the lines corresponding to the captured AST node, we're concatenating multiple formatted output lines per expected match here